sslcontext-kickstart/src/main/java/nl/altindag/ssl/util/KeyStoreUtils.java

```diff
@@ -257,15 +259,21 @@ public static List<KeyStore> loadSystemKeyStores() {
     @SuppressWarnings("SameParameterValue")
     static Optional<KeyStore> createKeyStoreIfAvailable(String keyStoreType, char[] keyStorePassword) {
         try {
-            KeyStore keyStore = createKeyStore(keyStoreType, keyStorePassword);
-
-            if (LOGGER.isDebugEnabled()) {
-                int totalTrustedCertificates = countAmountOfTrustMaterial(keyStore);
-                LOGGER.debug("Successfully loaded KeyStore of the type [{}] having [{}] entries", keyStoreType, totalTrustedCertificates);
+            return CompletableFuture.supplyAsync(() -> createKeyStore(keyStoreType, keyStorePassword))
+                    .thenApply(keyStore -> {
+                        if (LOGGER.isDebugEnabled()) {
+                            int totalTrustedCertificates = countAmountOfTrustMaterial(keyStore);
+                            LOGGER.debug("Successfully loaded KeyStore of the type [{}] having [{}] entries", keyStoreType, totalTrustedCertificates);
+                        }
+                        return keyStore;
+                    })
+                    .thenApply(Optional::of)
+                    .get(500, TimeUnit.MILLISECONDS);
```

Comment:
The timeout will make the CompletableFuture complete exceptionally with a TimeoutException, but the underlying thread will probably stay blocked.
After multiple calls, it may create a thread leak (supplyAsync uses the ForkJoin pool, which is not limited in the number of threads AFAIR).

If you want more control, you should not use CompletableFuture, but your own thread, and attempt to interrupt it after the timeout. But I am not even sure it would work, because the stacktrace showed that the process was blocked in native code (sun.security.mscapi.CKeyStore.loadKeysOrCertificateChains) and I have no idea if this could be interrupted.

Reply:
Yeah, you might be correct indeed. The underlying thread can still be active...
I will investigate whether the native code can be interrupted.
Thank you for reviewing this PR.

Reply:
Correcting myself :
> The timeout will make the CompletableFuture complete exceptionally with a TimeoutException

this is inaccurate. `get(timeout)` will throw a TimeoutException if the time is exceeded, but the CompletableFuture will still be not completed.

You could be tempted to do:
```
   try {
      future.get(timeout);
   } catch (TimeoutException e) {
      future.cancel(true);
   }
```
The problem is that the "interruptIfRunning" flag is not used by the completable future framework. And the question about the actual effect of interrupting a native method call remains anyway.